src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.File.Tests.cs

```diff
@@ -206,5 +208,84 @@ public void Add_SymbolicLink(TarEntryFormat format, bool createTarget)
                 Assert.Null(reader.GetNextEntry());
             }
         }
+
+        [Theory]
+        [MemberData(nameof(WriteEntry_CopyArchive_Data))]
+        public void WriteEntry_CopyArchive(TestTarFormat testFormat, string testCaseName, CompressionMethod compressionMethod, bool copyData)
+        {
+            TarEntryFormat entryFormat = GetEntryFormatForTestTarFormat(testFormat);
+            string pathWithExpectedFiles = GetTestCaseUnarchivedFolderPath(testCaseName);
+            if (!Path.EndsInDirectorySeparator(pathWithExpectedFiles))
+            {
+                pathWithExpectedFiles = pathWithExpectedFiles + Path.DirectorySeparatorChar;
+            }
+
+            using Stream file = GetTarMemoryStream(compressionMethod, testFormat, testCaseName);
+            using Stream originArchive = compressionMethod == CompressionMethod.GZip ? new GZipStream(file, CompressionMode.Decompress) : file;
+
+            VerifyCopyArchive(pathWithExpectedFiles, originArchive, entryFormat, copyData);
+        }
+
+        protected void VerifyCopyArchive(string pathWithExpectedFiles, Stream originArchive, TarEntryFormat entryFormat, bool copyData)
+        {
+            TarEntry entry;
+
+            using MemoryStream destinationArchive = new MemoryStream();
+            using (TarReader reader = new TarReader(originArchive, leaveOpen: false))
+            {
+                using (TarWriter writer = new TarWriter(destinationArchive, entryFormat, leaveOpen: true))
+                {
+                    while ((entry = reader.GetNextEntry(copyData)) != null)
+                    {
+                        if (entry is PaxTarEntry paxEntry)
+                        {
+                            paxEntry.GroupName = TestLongGName;
+                            paxEntry.UserName = TestLongUName;
+                        }
```

Comment:
Why does PaxTarEntry need to be special-cased?  What goes wrong if you remove this and the validation of it later in the test?

Reply:
PAX is the entry format affected by this bug. GroupName and UserName are among the extended attributes that are being affected by this fix, and those are properties that are only available in PAX and GNU. So I need to cast the object to be able to confirm their value was properly modified, because the GetNextEntry method returns the base class object TarEntry.

Reply:
Do we have a separate test that validates you can roundtrip between reader and writer without modification to the PAX entry?

Reply:
We have some tests that verify PAX extended attributes, but not for a round trip. I'll add them here. Thanks for mentioning them.

Reply:
Added some roundtrip verifications in the newest tests.